File: quixo/new_server/routes/product.routes.js
const express = require('express');
const router = express.Router();
const productController = require('../controllers/product.controller');

router.get('/all', productController.getAllProducts);
router.post('/all', productController.getAllProducts); // support both just in case

router.get('/id', productController.getProductById);
router.post('/id', productController.getProductById);

router.get('/recommended', productController.getRecommendedProducts);
router.post('/recommended', productController.getRecommendedProducts);

module.exports = router;

File: quixo/server/routes/product_routes.go
package routes

import (
	"quixo-server/controllers"

	"github.com/gin-gonic/gin"
)

func SetupProductRoutes(r *gin.Engine) {
	productGroup := r.Group("/api/product")
	{
		// Publicly fetchable routes that intelligently parse optional tokens inline
		productGroup.GET("/all", controllers.GetAllProducts)
		productGroup.GET("/id", controllers.GetProductByID)
		productGroup.GET("/recommended", controllers.GetRecommendedProducts)
	}
}
